### PR TITLE
GH-243 E2EC-1137 google map redraw when going to next page

### PR DESCRIPTION
**Pull request by waseemHybris**
GH-243

### CODE REVIEW

File: projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ const GOOGLE_API_KEY_PROPERTY_NAME = 'e2egoogleservices.apikey';
 const DEFAULT_SCALE = 12;
 const SELECTED_MARKER_SCALE = 16;
 
+
```
Comment (vkoulaxazov) [inline projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts]: Remove these extra blank lines

File: projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts [hunk 2]
```diff
@@ -40,24 +42,12 @@ export class GoogleMapRendererService {
             GOOGLE_MAP_API_URL,
             { key: result },
             () => {
-              this.initMap(mapElement, this.defineMapCenter(locations));
-              if (selectMarkerHandler) {
-                this.createMarkers(locations, selectMarkerHandler);
-              } else {
-                this.createMarkers(locations);
-              }
+              this.drawMap.call(this, mapElement, locations, selectMarkerHandler);
```
Comment (vkoulaxazov) [inline projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts]: invoke the method without the **call** operator

File: projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts [hunk 3]
```diff
@@ -40,24 +42,12 @@ export class GoogleMapRendererService {
...
     } else {
-      this.setMapOnAllMarkers(null);
-      if (selectMarkerHandler) {
-        this.createMarkers(locations, selectMarkerHandler);
-      } else {
-        this.createMarkers(locations);
-      }
-      this.googleMap.setCenter(this.defineMapCenter(locations));
-      this.googleMap.setZoom(DEFAULT_SCALE);
+      this.drawMap.call(this, mapElement, locations, selectMarkerHandler);
```
Comment (vkoulaxazov) [inline projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts]: same comment as above

File: projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts [hunk 4]
```diff
@@ -134,10 +124,16 @@ export class GoogleMapRendererService {
   }
 
   /**
-   * Moves all the markers to the given map
-   * @param map {@link google.maps.Map} the map where all the markers will be moved. Pass null if you just want to erase markers
+   * Initialize and draw the map
+   * @param locations array of locations to be displayed on the map
```
Comment (vkoulaxazov) [inline projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts]: you're missing a param for the HTMLElement, we also usually don't put comments in this project

File: projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts [hunk 5]
```diff
@@ -134,10 +124,16 @@ export class GoogleMapRendererService {
...
+   * @param selectMarkerHandler function to handle whenever a marker on a map is clicked
    */
-  private setMapOnAllMarkers(map: google.maps.Map): void {
-    this.markers.forEach(marker => marker.setMap(map));
+  drawMap(mapElement: HTMLElement, locations: any[], selectMarkerHandler: Function) {
```
Comment (vkoulaxazov) [inline projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts]: this method must be private if it isn't called by an outside class

File: projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts [hunk 6]
```diff
@@ -134,10 +122,21 @@ export class GoogleMapRendererService {
...
    */
-  private setMapOnAllMarkers(map: google.maps.Map): void {
-    this.markers.forEach(marker => marker.setMap(map));
+  private drawMap(
+    mapElement: HTMLElement,
+    locations: any[],
+    selectMarkerHandler: Function
+  ) {
+    this.initMap(mapElement, this.defineMapCenter(locations));
+    if (selectMarkerHandler) {
```
Comment (Andrey-Vlasov) [inline projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts]: I think this condition is redundant as the very same verification is done inside createMarkers function. And to my mind createMarkers function is the only place where it should be done.

**Pull request merged**